packages/middleware/__tests__/unit/apiClientFactory/apiClientFactory.spec.ts

```diff
@@ -106,7 +108,7 @@ describe("apiClientFactory", () => {
     const { createApiClient } = apiClientFactory<any, any>(params as any);
     const extensions = (createApiClient as any)._predefinedExtensions;
 
-    const apiWithContext: any = createApiClient.bind({
+    const apiWithContext: any = await createApiClient.bind({
```

Comment:
> Since you're calling this function immediately anyway, perhaps you could replace .bind()  with [.call()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call)?

same here